docs/howto/custom-shell.txt

```diff
@@ -0,0 +1,46 @@
+====================================== 
+How to customize the ``shell`` command 
+======================================
+
+To customize the :djadmin:`shell`, create a new :doc:`management command
+</howto/custom-management-commands>` which subclasses 
+``django.core.management.commands.shell.Command`` and overrides the existing
+``shell`` management command. See :ref:`Overriding commands 
+<overriding-commands>` for more details. 
+
+Customizing the auto-imports
+============================
+
+.. versionadded:: 5.2
+
+To customize the auto-importing behavior of :djadmin:`shell`, you can override
+the ``get_namespace()`` method. For example, assuming you have a ``polls``
+application in :setting:`INSTALLED_APPS`:
+
+.. code-block:: python
+    :caption: ``polls/management/commands/shell.py``
+
+    from django.core.management.commands import shell
+
+
+    class Command(shell.Command):
+        def get_namespace(self):
+            from django.urls.base import resolve, reverse
+
+            return {
+                **super().get_namespace(),
+                "resolve": resolve,
+                "reverse": reverse,
+            }
+
+This way, your customized shell will import models from your apps along with the
+``resolve`` and ``reverse`` methods, allowing you to use them without needing
+additional imports.
+
+.. note::
+
+    If you prefer your shell without any auto-imported model, you can omit
+    calling the ``**super().get_namespace()`` method when returning the
+    namespace, as it can be empty.
+
+
```

Comment:
I think it might be better just to have a second example
```suggestion
This way, your custom shell command will import models from your apps along with
the ``resolve`` and ``reverse`` methods. You can then use them without needing
additional imports.

If you prefer not to have any auto-imported models. You can omit calling
``**super().get_namespace()`` in the ``get_namespace()`` method:

.. code-block:: python
    :caption: ``polls/management/commands/shell.py``

    from django.core.management.commands import shell


    class Command(shell.Command):
        def get_namespace(self):
            return {}

```